scipy/stats/_continuous_distns.py

```diff
@@ -2900,6 +2900,129 @@ def _entropy(self, c):
 genhalflogistic = genhalflogistic_gen(a=0.0, name='genhalflogistic')
 
 
+class genhyperbolic_gen(rv_continuous):
+    r"""A generalized hyperbolic continuous random variable.
+
+    %(before_notes)s
+
+    Notes
+    -----
+    The probability density function for `genhyperbolic` is:
+
+    .. math::
+
+        f(x, \lambda, \alpha, \beta, \delta, \mu) = 
+           \frac{(\gamma/\delta)^\lambda}{\sqrt{2\pi}K_\lambda(\delta \gamma)}
+           e^{\beta (x - \mu)} \times \frac{K_{\lambda - 1/2}
+           (\alpha \sqrt{\delta^2 + (x - \mu)^2})}
+           {(\sqrt{\delta^2 + (x - \mu)^2} / \alpha)^{1/2 - \lambda}}
+
+    for :math:`x \in (-\infty; \infty)`,
+    and :math:`\lambda, \mu \in \mathbb{R}`
+    ans :math: `0 \le |\beta| < \alpha ` \beta,
+    and :math: `\delta \in \mathbb{R}_{>0}`.
+    
+    Where: :math:`\gamma := \sqrt{\alpha^2-\beta^2}`.
+    :math: `K_{\lambda}(.)` denotes the modified Bessel function of the third
+    kind and order :math: `\lambda` (`scipy.special.kn`)
+
+    `genhyperbolic` takes ``lmbda`` as a tail parameter for :math:`\lambda`.
+    takes ``alpha`` as a shape parameter for :math:`\alpha`.
+    takes ``beta`` as a skeweness parameter for :math:`\beta`.
+    takes ``delta`` as a scale parameter for :math:`\delta`.
+    takes ``mu`` as a location parameter for :math:`\mu`.
+
+
+    %(after_notes)s
+
+    References
+    ----------
+    O. Barndorff-Nielsen, "Hyperbolic Distributions and Distributions on
+    Hyperbolae", Scandinavian Journal of Statistics, Vol. 5(3),
+    pp. 151-157, 1978.
+
+    Scott, David J, Würtz, Diethelm, Dong, Christine and Tran, Thanh Tam, (2009),
+    Moments of the generalized hyperbolic distribution, MPRA Paper,
+    University Library of Munich, Germany, https://EconPapers.repec.org/RePEc:pra:mprapa:19081.
+
+    %(example)s
+
+    """
+    
+    def _argcheck(self, lmbda, alpha, beta, delta, mu):
+        
+        @np.vectorize
+        def argcheck_single(lmbda, alpha, beta, delta, mu):
+            if lmbda > 0:
+                return np.all((np.absolute(beta) < alpha) & (delta >= 0))
+            elif lmbda == 0:
+                return np.all((np.absolute(beta) < alpha) & (delta > 0))
+            else:
+                return np.all((np.absolute(beta) <= alpha) & (delta > 0))
+                
+        return argcheck_single(lmbda, alpha, beta, delta, mu)
+
+    def norming_constant(self, lmbda=1, alpha=1, beta=0, delta=1):
```

Comment:
```suggestion
    def _norming_constant(self, lmbda=1, alpha=1, beta=0, delta=1):
```
Private methods should start with `_`.